conans/client/package_layouts/package_base_layout.py

```diff
@@ -0,0 +1,19 @@
+# coding=utf-8
+
+import os
+
+from conans.model.ref import ConanFileReference
+
+
+class PackageBaseLayout(object):
+    """ User customizable layout for packages """
+
+    def __init__(self, base_folder, conan_ref, short_paths=False):
+        assert isinstance(conan_ref, ConanFileReference)
+        self._conan_ref = conan_ref
```

Comment:
A generic layout might not be bounded to a conan_ref (local package without user/channel yet).

Reply:
We need a full reference. We will make editable a package given its full reference. If it is not in the conanfile.py, the user will make it explicit through the command line. I think it is needed.